crates/scope/src/builder.rs

```diff
@@ -2017,6 +2034,16 @@ impl ScopeBuilderStack {
                 let inner_base = inner.base();
                 let outer_base = outer.base_mut();
 
+                match self.closed_over_bindings_for_lazy.last_mut() {
+                    Some(bindings) => {
+                        for name in inner_base.name_tracker.defined_and_closed_over_vars() {
+                            bindings.push(Some(*name));
+                        }
+                        bindings.push(None);
+                    }
+                    None => {}
```

Comment:
Should we panic here, unless this is to handle the global case?
If so could we assert that this is the global scope?

Reply:
it's popping lexical scope in top-level script.
we could assert that there's no functionscope in the stack tho, it's not simple assertion.